examples/regexp-find-operations/regexp_find_operations.bal

```diff
@@ -0,0 +1,55 @@
+import ballerina/io;
+import ballerina/lang.regexp;
+
+public function main() {
+    string logContent = string `
+        2024-09-19 10:02:01 WARN  [UserLogin] - Failed login attempt for user: johndoe
+        2024-09-19 10:03:17 ERROR [Database] - Connection to database timed out
+        2024-09-19 10:04:05 WARN  [RequestHandler] - Response time exceeded threshold for /api/v1/users
+        2024-09-19 10:05:45 INFO  [Scheduler] - Scheduled task started: Data backup
+        2024-09-19 10:06:10 ERROR [Scheduler] - Failed to start data backup: Permission denied
+        2024-09-19 10:11:55 INFO  [Security] - Security scan completed, no issues found
+        2024-09-19 10:12:30 ERROR [RequestHandler] - 404 Not Found: /api/v1/products`;
+
+    // Regex to match error logs with three groups:
+    // 1. Timestamp (e.g., 2024-09-19 10:03:17).
+    // 2. Component (e.g., Database, Scheduler).
+    // 3. Log message (e.g., Connection to database timed out).
+    string:RegExp errorLogPattern = re `(\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}) ERROR \[(\w+)\] - (.*)`;
+
+    // Retrieving the first error log from the `logContent`.
+    regexp:Span firstErrorLog = <regexp:Span>errorLogPattern.find(logContent);
+    io:println(string `First error log: ${firstErrorLog.substring()}`);
```

Comment:
Wouldn't it be better to properly handle the `()` case (instead of casting)? It should show users why it could be null
```suggestion
    regexp:Span? firstErrorLog = errorLogPattern.find(logContent);
    if fristErrorLog == () {
        io:println("failed to find first error log");
        return;
    }
   io:println(string `First error log: ${firstErrorLog.substring()}`);
```